src/Elastic.Apm.AspNetCore/Extensions/TransactionExtentions.cs

```diff
@@ -13,26 +14,34 @@ internal static class TransactionExtensions
 		/// </summary>
 		/// <param name="transaction">Transaction object</param>
 		/// <param name="isForError">Is request body being captured for error (otherwise it's for transaction)</param>
-		/// <param name="request">Current http request</param>
+		/// <param name="httpRequest">Current http context containing the http request</param>
 		/// <param name="logger">Logger object</param>
-		internal static void CollectRequestBody(this Transaction transaction, bool isForError, HttpRequest request, IApmLogger logger)
+		internal static async Task CollectRequestBody(this Transaction transaction, bool isForError, HttpRequest httpRequest, IApmLogger logger)
 		{
 			if (!transaction.IsSampled) return;
 
+			if(httpRequest == null) return;
```

Comment:
IMHO we should avoid passing `null` as much as possible - it will make it easier for us to use C# 8 Nullable Reference Types (hopefully in the near future). It's worth checking if this method is ever called when `httpRequest` is `null`.

Reply:
This parameter comes from the outside 

```
public async Task InvokeAsync(HttpContext context)
```

And we pass it like:

```
await transaction.CollectRequestBody(true, context.Request, _logger)
```

In general, the `HttpRequest` will always come from outside, it's not something we control, and I expect `HttpContext.Request` being nullable. 

I'd feel better with the check. I hope this is not a blocker.